plugins/transport/socket/main.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"sync"
+
+	"github.com/infrawatch/apputils/logging"
+	"github.com/infrawatch/sg-core-refactor/pkg/config"
+	"github.com/infrawatch/sg-core-refactor/pkg/data"
+	"github.com/infrawatch/sg-core-refactor/pkg/transport"
+)
+
+const maxBufferSize = 4096
+
+type configT struct {
+	Address string `validate:"required"`
+}
+
+//Socket basic struct
+type Socket struct {
+	conf   configT
+	logger *logging.Logger
+}
+
+//Run implements type Transport
+func (s *Socket) Run(ctx context.Context, wg *sync.WaitGroup, w transport.WriteFn, done chan bool) {
+	defer wg.Done()
+
+	msgBuffer := make([]byte, maxBufferSize)
+	var laddr net.UnixAddr
+
+	laddr.Name = s.conf.Address
+	laddr.Net = "unixgram"
+
+	os.Remove(s.conf.Address)
+
+	pc, err := net.ListenUnixgram("unixgram", &laddr)
+	if err != nil {
+		s.logger.Metadata(logging.Metadata{"plugin": "socket", "error": err})
+		s.logger.Error("failed to listen on unix soc")
+		return
+	}
+
+	s.logger.Metadata(logging.Metadata{"plugin": "socket"})
+	s.logger.Info(fmt.Sprintf("socket listening on %s", laddr.Name))
+	go func() {
+		for {
+			n, err := pc.Read(msgBuffer)
+			//fmt.Printf("received message: %s\n", string(msgBuffer))
+
+			if err != nil || n < 1 {
+				done <- true
```

Comment:
Looking to see if anything generates an error in this case. Does this survive multiple open/closes from the bridge?

Reply:
I'm haven't tested it. This is the condition used by the original sg-core

Reply:
This needs to be wrapped in error handling code for socket closures as Chris mentioned...